packages/webpack-inline-constant-exports-plugin/index.js

```diff
@@ -0,0 +1,117 @@
+const HarmonyImportSideEffectDependency = require( 'webpack/lib/dependencies/HarmonyImportSideEffectDependency' );
+const HarmonyImportSpecifierDependency = require( 'webpack/lib/dependencies/HarmonyImportSpecifierDependency' );
+const ConstDependency = require( 'webpack/lib/dependencies/ConstDependency' );
+
+const moduleTypes = [ 'javascript/auto', 'javascript/dynamic', 'javascript/esm' ];
+
+function addConstantExport( module, name, value ) {
+	if ( ! module.constantExports ) {
+		module.constantExports = {};
+	}
+	module.constantExports[ name ] = value;
+}
+
+function removeDependency( module, dep ) {
+	module.removeDependency( dep );
+	dep.module.reasons = dep.module.reasons.filter( r => r.dependency !== dep );
+}
+
+class InlinePlugin {
+	constructor( matchers ) {
+		if ( Array.isArray( matchers ) ) {
+			this.matchers = matchers;
+		} else {
+			this.matchers = [ matchers ];
+		}
+	}
+
+	isConstantsModule( module ) {
+		return module && this.matchers.some( matcher => matcher.test( module.resource ) );
+	}
+
+	apply( compiler ) {
+		compiler.hooks.compilation.tap( 'InlinePlugin', ( compilation, { normalModuleFactory } ) => {
+			const handleParser = parser => {
+				const handleExport = ( statement, declaration ) => {
```

Comment:
It appears that what you consider to be constant dependencies are `const`s and literal default exports only, right? That sounds fine, just wanted to confirm :)

Reply:
➕ on choosing `const` and `literal`s only

➕ to what @sirreal says below - seems like we should make sure we're sticking to primitive literals. astexplorer.net shows `[]` to be an `ArrayExpression` and `{}` to be an `ObjectExpression` so I think we're okay with `Literal`, but I don't know what webpack is doing

Reply:
> I don't know what webpack is doing

webpack uses the Acorn parser (not the Babel's Babylon one). The AST types are the same in this case for both parsers. Different parsers usually only differ in small details, not fundamental things like AST types for common constructs.